pkg/metrics/process/sampler_darwin.go

```diff
@@ -86,23 +88,27 @@ func (ps *processSampler) Sample() (results sample.EventBatch, err error) {
 		return nil, err
 	}
 
-	var containerDecorator metrics.ProcessDecorator
-	if ps.containerSampler.Enabled() {
-		containerDecorator, err = ps.containerSampler.NewDecorator()
-		if err != nil {
-			// ensure containerDecorator is set to nil if error
-			containerDecorator = nil
-			if id := containerIDFromNotRunningErr(err); id != "" {
-				if _, ok := containerNotRunningErrs[id]; !ok {
-					containerNotRunningErrs[id] = struct{}{}
+	var containerDecorators []metrics.ProcessDecorator
+
+	for _, containerSampler := range ps.containerSamplers {
+		if containerSampler.Enabled() {
+			decorator, err := containerSampler.NewDecorator()
+			if err != nil {
+				// ensure containerDecorator is set to nil if error
+				decorator = nil
+				if id := containerIDFromNotRunningErr(err); id != "" {
+					if _, ok := containerNotRunningErrs[id]; !ok {
+						containerNotRunningErrs[id] = struct{}{}
+						mplog.WithError(err).Warn("instantiating container sampler process decorator")
+					}
+				} else {
 					mplog.WithError(err).Warn("instantiating container sampler process decorator")
-				}
-			} else {
-				mplog.WithError(err).Warn("instantiating container sampler process decorator")
-				if strings.Contains(err.Error(), "client is newer than server") {
-					mplog.WithError(err).Error("Only docker api version from 1.24 upwards are officially supported. You can still use the docker_api_version configuration to work with older versions. You can check https://docs.docker.com/develop/sdk/ what api version maps with each docker version.")
+					if strings.Contains(err.Error(), "client is newer than server") {
+						mplog.WithError(err).Error("Only docker api version from 1.24 upwards are officially supported. You can still use the docker_api_version configuration to work with older versions. You can check https://docs.docker.com/develop/sdk/ what api version maps with each docker version.")
+					}
 				}
 			}
+			containerDecorators = append(containerDecorators, decorator)
```

Comment:
Should we check that `decorator != nil`?

Reply:
it's modified in last commit https://github.com/newrelic/infrastructure-agent/commit/337fedfde8783cd63ae68c419a492281d32a517a